libs/hdf-converters/src/msft-secure-score-mapper.ts

```diff
@@ -0,0 +1,203 @@
+                }
+              }
+            },
+            desc: {path: 'description'},
+            impact: {
+              transformer: (data: ControlScore) => {
+                // return controlCategory from the profile document where its id matches the controlName
+                const knownMaxScores = this.getProfiles(
+                  data.controlName || ''
+                ).map((profile) => profile.maxScore || 0);
+
+                if (knownMaxScores.length === 0) {
+                  return 0.5;
+                }
+
+                const highMaxScore = Math.max(...knownMaxScores);
+                return highMaxScore / 10.0;
+              }
+            },
+            refs: [],
+            tags: {
+              group: {
+                transformer: (data: ControlScore) => {
+                  // return controlCategory from the profile document where its id matches the controlName
+                  return this.getProfiles(data.controlName || '').map(
+                    (profile) => profile.controlCategory
+                  );
+                }
+              },
+              tiers: {
+                transformer: (data: ControlScore) => {
+                  // return tiers from the profile document where its id matches the controlName
+                  return this.getProfiles(data.controlName || '').map(
+                    (profile) => profile.tier
+                  );
+                }
+              },
+              threats: {
+                transformer: (data: ControlScore) => {
+                  // return unique threats from the profile document where its id matches the controlName
+                  const uniqs: Set<string> = new Set();
+                  this.getProfiles(data.controlName || '').forEach(
+                    (profile: SecureScoreControlProfile) =>
+                      profile.threats?.forEach((threat) => uniqs.add(threat))
+                  );
+                  return [...uniqs];
+                }
+              },
+              services: {
+                transformer: (data: ControlScore) => {
+                  // return thrserviceeats from the profile document where its id matches the controlName
+                  return this.getProfiles(data.controlName || '').map(
+                    (profile) => profile.service
+                  );
+                }
+              },
+              userImpacts: {
+                transformer: (data: ControlScore) => {
+                  // return userImpacts from the profile document where its id matches the controlName
+                  return this.getProfiles(data.controlName || '')
+                    .filter((profile) => profile.userImpact !== undefined)
+                    .map((profile) => profile.userImpact);
+                }
+              },
+              nist: ['SA-11', 'RA-5']
+            },
+            source_location: {},
+            descriptions: [
```

Comment:
did not address @ejaronne's request for a rationale subdescription based off of profile.remediationimpact

Reply:
implemented